docs/configuration/configuration.md

```diff
@@ -87,6 +87,7 @@ An example [oauth2-proxy.cfg]({{ site.gitweb }}/contrib/oauth2-proxy.cfg.example
 | `--provider` | string | OAuth provider | google |
 | `--provider-display-name` | string | Override the provider's name with the given string; used for the sign-in page | (depends on provider) |
 | `--ping-path` | string | the ping endpoint that can be used for basic health checks | `"/ping"` |
+| `--ping-user-agent` | string | a User-Agent that can be used for basic health checks | `""` (don't check user agent) |
```

Comment:
This should probably mention that it will be used for log filtering 

Any reason not to make this a logging option that excludes particular user agents? That way users could specify multiple user agents should they wish to (eg Azure LB and K8s health checks)

Reply:
It's not just for logging. A User-Agent match forces it to only serve health requests (it's impossible to serve a regular request if the user agent matches for instance). Logging of those requests is governed by the `--silence-ping-logging` option, not whether it's using a path or a user-agent to identify it as such.